apps/managedidentity/managedidentity_test.go

```diff
@@ -1131,3 +1132,95 @@ func TestCreatingIMDSClient(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkAcquireTokenSilentLatency(b *testing.B) {
+	mockClient := mock.Client{}
+	resource := "https://graph.microsoft.com"
+
+	responseBody, err := getSuccessfulResponse(resource, true)
+	if err != nil {
+		b.Fatalf(errorFormingJsonResponse, err.Error())
+	}
+
+	mockClient.AppendResponse(mock.WithHTTPStatusCode(http.StatusOK), mock.WithBody(responseBody))
+
+	before := cacheManager
+	defer func() { cacheManager = before }()
+	cacheManager = storage.New(nil)
+
+	client, err := New(SystemAssigned(), WithHTTPClient(&mockClient))
+	if err != nil {
+		b.Fatal(err)
+	}
+	ctx := context.Background()
+	_, err = client.AcquireToken(ctx, resource)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	var latencies []time.Duration
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		start := time.Now()
+		_, err := client.AcquireToken(ctx, resource)
+		if err != nil {
+			b.Fatal(err)
+		}
+		latencies = append(latencies, time.Since(start))
+	}
+
+	sort.Slice(latencies, func(i, j int) bool { return latencies[i] < latencies[j] })
+
+	p50 := latencies[len(latencies)*50/100]
+	p90 := latencies[len(latencies)*90/100]
+	p99 := latencies[len(latencies)*99/100]
+
+	b.ReportMetric(float64(p50.Microseconds()), "p50_us")
+	b.ReportMetric(float64(p90.Microseconds()), "p90_us")
+	b.ReportMetric(float64(p99.Microseconds()), "p99_us")
```

Comment:
Are these values interesting? The benchmark covers getting a token from the cache i.e., it measures how long it took to acquire an uncontested lock or two and find a value in a map that never grows. I imagine the distribution is therefore quite flat

Reply:
I will create a separate PR for performance test.